### PR TITLE
chore(build): establish an OT-3 spin of the app

### PR DESCRIPTION
**Pull request by sfoster1**
During the development process of the OT-3, we want to have a separate release channel for software for the under-development machine. This will let us keep everybody internally in sync, without bothering users of the OT-2 with updates containing no new functionality for their machines, and without making us QA the software for the OT-2.

This PR adds that by
- Creating a build variant of the app called the "ot-3 app", which is versioned based on tags matching `ot3@*`
- Deploying this build variant to a separate github bucket and configuring the build variant to look to that github bucket for its updates
- (slightly) reskinning the app to make it more apparent that it's the under-development version

By changing the electron app-id, this app can be installed side by side with the production opentrons app. it gets a different app and installer icon, a different window name, and so on. However, it runs the same code and has basically unchanged functionaltiy from the production app.

Please take a look at the changes in electron-builder.config.js and the makefiles, and pay particular attention to the changes in the app-build-test workflow.

Also let me know whether what I've done to change the navbar logo is a good idea:
- I would like this to be a fully separate spin of the app with differences baked in at build, because the update details (which are the most important part) have to be; therefore, the logo is determined at build
- So I used a webpack define, which I know is gross

Closes RCORE-190

### CODE REVIEW

File: .github/workflows/app-test-build-deploy.yaml [hunk 1]
```diff
@@ -184,5 +264,54 @@ jobs:
...
+        with:
+          path: .
+      - name: 'configure ot3 s3 deploy creds'
+        run: |
+          aws configure set aws_access_key_id ${{ secrets.S3_OT3_APP_DEPLOY_KEY_ID }} --profile identity
+          aws configure set aws_secret_access_key ${{ secrets.S3_OT3_APP_DEPLOY_SECRET }} --profile identity
+          aws configure set region us-east-2 --profile identity
+          aws configure set output json --profile identity
+          aws configure set region us-east-2 --profile deploy
+          aws configure set role_arn arn:aws:iam::879285218407:role/administrator --profile deploy
```
Comment (shlokamin) [inline .github/workflows/app-test-build-deploy.yaml]: lol this looks shady

Comment (shlokamin) [reply .github/workflows/app-test-build-deploy.yaml]: should we be exposing this id here?

File: app-shell/electron-builder.config.js [hunk 2]
```diff
@@ -8,15 +8,39 @@ const USE_PYTHON = process.env.NO_PYTHON !== 'true'
 const NO_USB_DETECTION = process.env.NO_USB_DETECTION === 'true'
 const project = process.env.OPENTRONS_PROJECT ?? 'robot-stack'
 
+const ot3PublishConfig =
+  OT_APP_DEPLOY_BUCKET && OT_APP_DEPLOY_FOLDER
+    ? {
+        provider: 'generic',
```
Comment (shlokamin) [inline app-shell/electron-builder.config.js:14]: 🤘🏾

File: app-shell/electron-builder.config.js [hunk 3]
```diff
@@ -8,15 +8,39 @@ const USE_PYTHON = process.env.NO_PYTHON !== 'true'
...
     '**/*',
     'build/br-premigration-wheels',
     '!Makefile',
     '!python',
-    NO_USB_DETECTION ? '!node_modules/usb-detection' : '',
+    // this detail with node modules makes sure that any byproducts from the build process
+    // of the usb-detection submodule - which should only exist anyway if we don't get a
+    // prebuild - won't get packed into the app. If they are, they'll prevent the app
+    // from running on OSX, because they intern paths on the action runner in their binaries
+    // sometimes and gatekeeper will scan them, notice, and fail us.
```
Comment (shlokamin) [inline app-shell/electron-builder.config.js]: we should be able to clean this up when my freaking app shell odd work lands

**Review by shlokamin — APPROVED** [review]
thx for doing this, i think this is reasonable for now and we can update this as we evolve our deployment process

File: .github/workflows/app-test-build-deploy.yaml (hunk 1 above)
Comment (sfoster1) [reply .github/workflows/app-test-build-deploy.yaml]: i'll get rid

**Pull request merged**